plugins/out_s3/s3.c

```diff
@@ -1243,8 +1322,23 @@ static int put_all_chunks(struct flb_s3 *ctx)
                                 chunk->create_time, buffer, buffer_size);
             flb_free(buffer);
             if (ret < 0) {
-                s3_store_file_unlock(chunk);
                 chunk->failures += 1;
+                if (is_startup == FLB_TRUE) {
+                    if (chunk->failures > ctx->ins->retry_limit) {
+                        s3_retry_warn(ctx, (char *) fsf->meta_buf, NULL,
+                                      chunk->create_time, FLB_FALSE);
+                        s3_store_file_delete(ctx, chunk);
+                    }
+                    else {
+                        s3_retry_warn(ctx, (char *) fsf->meta_buf, NULL,
+                                      chunk->create_time, FLB_TRUE);
+                        s3_store_file_unlock(chunk);
+                    }
+                }
+                else {
+                    flb_plg_error(ctx->ins, "Failed to flush chunk tag=%s",
+                                  (char *) fsf->meta_buf);
```

Comment:
This code doesn't look like what we discussed to me. Also has it changed recently?

Didn't we agree that:

1. For the startup case, we want to use the s3_retry_warn. Since the chunk can be retried. Please note that technically the overlimit case should not be able to happen on startup. The chunks were just picked up from filesystem and so they can't have yet accumulated any retries for this invocation of Fluent Bit. 
2. For the shutdown case, we can't retry, but we give the user a message like "Chunk will remain buffered on the filesystem in the store_dir". So the user knows the data can be picked back up when FLB restarts. I swear I remember you had this message here like last week- can we please have this.